Source/WebKit/WebProcess/GPU/media/AudioVideoRendererRemote.h

```diff
@@ -210,50 +210,57 @@ class AudioVideoRendererRemote final
     WTFLogChannel& logChannel() const final;
 #endif
 
+    void setState(RemoteAudioVideoRendererState);
+
     void ensureOnDispatcher(Function<void()>&&);
-    void ensureOnDispatcherSync(Function<void()>&&);
+    void ensureOnDispatcherSync(NOESCAPE Function<void()>&&);
+    void ensureOnDispatcherWithConnection(Function<void(AudioVideoRendererRemote&, IPC::Connection&)>&&);
     static WorkQueue& queueSingleton();
     bool isGPURunning() const { return !m_shutdown; }
 
     void updateCacheState(const RemoteAudioVideoRendererState&);
+    static constexpr size_t kMaxPendingSample = 10;
+    struct ReadyForMoreData {
+        bool isReadyForMoreData() const { return pendingSamples < kMaxPendingSample; }
+        size_t pendingSamples { 0 };
+    };
+    ReadyForMoreData& readyForMoreData(TrackIdentifier);
```

Comment:
I would be tempted to add methods like clearReadyForMoreData/incrementReadyForMoreData instead of passing a ref that is only valid until the lock is kept.